components/sensor/ld2420.rst

```diff
@@ -0,0 +1,261 @@
+.. _ld2420-component:
+
+The ``ld2420`` sensor platform allows you to use the HLK-LD2420 motion and presence sensor.
+The :ref:`UART <uart>` is required to be set up in your configuration for this sensor to work.
+
+Use of hardware UART pins is critical in order to support the baud rates of the LD2420 sensor.
+Firmware v1.5.3 and newer default to 115200 baud, 256000 baud on all older firmwares.
+
+.. figure:: /images/ld2420.jpg
+    :align: center
+    :width: 50.0%
+
+    HLK-LD2420 Low power motion and presence sensor.
+
+Pinouts
+-------
+
+.. list-table:: Firmware 1.5.2 and older
+    :widths: 25 25 25
+    :header-rows: 1
+
+    * - Pin#
+      - Name
+      - Function
+    * - 1
+      - 3v3
+      - VCC
+    * - 2
+      - GND
+      - GND
+    * - 3
+      - OT1
+      - Presence Signal Output
+    * - 4
+      - RX
+      - Serial Rx (to ESP Tx)
+    * - 5
+      - OT2
+      - Serial Tx (to ESP Rx)
+
+.. list-table:: Firmware 1.5.3 and newer
+    :widths: 25 25 25
+    :header-rows: 1
+
+    * - Pin#
+      - Name
+      - Function
+    * - 1
+      - 3v3
+      - VCC
+    * - 2
+      - GND
+      - GND
+    * - 3
+      - OT1
+      - Serial Tx (to ESP Rx)
+    * - 4
+      - RX
+      - Serial Rx (to ESP Tx)
+    * - 5
+      - OT2
+      - Presence Signal Output
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      id: ld2420_uart
+      tx_pin: REPLACEME
+      rx_pin: REPLACEME
+      baud_rate: 115200
+      parity: NONE
+      stop_bits: 1
+
+    # The LD2420 has 16 time gate intervals (0-15) with a gate
+    # resolution of 0.75 meters
+    ld2420:
+      presence_time_window: 120s
+      detection_gate_min: 1
+      detection_gate_max: 12
+      g0_move_threshold: 60000
+      g0_still_threshold: 40000
+      g1_move_threshold: 30000
+      g1_still_threshold: 20000
+      g2_move_threshold: 400
+      g2_still_threshold: 200
+      g3_move_threshold: 250
+      g3_still_threshold: 200
+      g4_move_threshold: 250
+      g4_still_threshold: 200
+      g5_move_threshold: 250
+      g5_still_threshold: 200
+      g6_move_threshold: 250
+      g6_still_threshold: 200
+      g7_move_threshold: 250
+      g7_still_threshold: 150
+      g8_move_threshold: 250
+      g8_still_threshold: 150
+      g9_move_threshold: 250
+      g9_still_threshold: 100
+      g10_move_threshold: 250
+      g10_still_threshold: 100
+      g11_move_threshold: 250
+      g11_still_threshold: 100
+      g12_move_threshold: 250
+      g12_still_threshold: 100
+      g13_move_threshold: 200
+      g13_still_threshold: 100
+      g14_move_threshold: 200
+      g14_still_threshold: 100
+      g15_move_threshold: 200
+      g15_still_threshold: 100
+
+.. note::
+
+    For UART configuration, ``baud_rate``, ``parity`` and ``stop_bits`` **must be** respectively ``115200 or 256000``, ``NONE`` and ``1``.
+    Use of hardware UART pins is critical in order to support the out-of-the-box 256000 or 115200 baud rates of the LD2420 sensor.
```

Comment:
Given the issues I see already mentioned on the PR, it might be better to make this a `warning` and move the bit about the hardware UART to the top, like this:

```suggestion
.. warning::

    Use of a hardware UART is critical in order to support the out-of-the-box 256000 or 115200 baud rates of the LD2420 sensor. Communication problems are likely to result otherwise.
    In addition, for UART configuration, ``baud_rate``, ``parity`` and ``stop_bits`` **must be** respectively ``115200 or 256000``, ``NONE`` and ``1``.
```